aiutils/tftools/layers.py

```diff
@@ -81,3 +81,41 @@ def conv2d(input,
                                    strides=strides,
                                    padding=padding) + b)
     return output
+
+
+def batch_norm(input, name):
+    """ Batch norm layer helper.
+    
+    Gets mean and variance, and creates offset and scale parameters with good initial values.
+    Then applies the batch_normalization op.
+    
+    Args:
+      input (tensor): Input to the layer. 
+        Should have shape `[batch, in_dim]` or `[batch, in_height, in_width, in_dim]`.
+        Must be one of the following types: `float32`, `float64`.
+      name (str): Name used by the `tf.variable_scope`.
+      
+    Returns:
+      output (tensor): Batch normalized activations.
+        Will have the same shape as input.
+    """
+    rank = len(input.get_shape().as_list())
+    in_dim = input.get_shape().as_list()[-1]
+    if rank == 2:
+        axes = [0]
+    elif rank == 4:
+        axes = [0, 1, 2]
+    else:
+        raise ValueError('rank must be 2 or 4.')
```

Comment:
I think the idea is to make clear what your assumptions about inputs are as soon as possible, rather than have a user read through the logic and then understand it indirectly. When computation isn't that expensive, readability should be given preference.  Though here it doesn't seem to be that big a deal either way.